qoqo/tests/integration/operations/analog_operations.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright © 2021-2023 HQS Quantum Simulations GmbH. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the
+// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+
+use pyo3::prelude::*;
+use pyo3::Python;
+use qoqo::operations::convert_operation_to_pyobject;
+use qoqo::operations::{
+    ApplyConstantSpinHamiltonianWrapper, ApplyTimeDependentSpinHamiltonianWrapper,
+};
+use qoqo_calculator::CalculatorFloat;
+use roqoqo::operations::Operation;
+use roqoqo::operations::*;
+#[cfg(feature = "json_schema")]
+use roqoqo::ROQOQO_VERSION;
+
+use std::collections::HashMap;
+use struqture::prelude::*;
+use struqture::spins::{PauliProduct, SpinHamiltonian};
+use struqture_py::spins::SpinHamiltonianSystemWrapper;
+
+fn new_system(py: Python, number_spins: Option<usize>) -> &PyCell<SpinHamiltonianSystemWrapper> {
+    let system_type = py.get_type::<SpinHamiltonianSystemWrapper>();
+    system_type
+        .call1((number_spins,))
+        .unwrap()
+        .downcast::<PyCell<SpinHamiltonianSystemWrapper>>()
+        .unwrap()
+}
+
+/// Test new() function for ApplyConstantSpinHamiltonian
+#[test]
+fn test_new_constantspinhamiltionian() {
+    let pp = PauliProduct::new().z(0);
+    let mut hamiltonian = SpinHamiltonian::new();
+    hamiltonian
+        .add_operator_product(pp.clone(), CalculatorFloat::from(1.0))
+        .unwrap();
+
+    let input_operation = Operation::from(ApplyConstantSpinHamiltonian::new(
+        hamiltonian.clone(),
+        (0.1).into(),
+    ));
+    let method = "__eq__";
+    let operation = convert_operation_to_pyobject(input_operation).unwrap();
+    pyo3::prepare_freethreaded_python();
+    Python::with_gil(|py| {
+        let number_spins: Option<usize> = None;
+        let new_system = new_system(py, number_spins);
+        new_system
+            .call_method1("add_operator_product", ("0Z", 1.))
+            .unwrap();
+        let system_wrapper = new_system
+            .extract::<SpinHamiltonianSystemWrapper>()
+            .unwrap();
```

Comment:
Are these lines necessary for the test to run?

Reply:
I thought so, because `system_wrapper` is an input to create any analog operation object. Is there a better way to do this?